packages/flutter/lib/src/material/material.dart

```diff
@@ -548,76 +543,77 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   }
 }
 
-class _RenderInkFeatures extends RenderProxyBox implements MaterialInkController {
-  _RenderInkFeatures({
-    RenderBox? child,
-    required this.vsync,
-    required this.absorbHitTest,
-    this.color,
-  }) : super(child);
+/// A [RenderBox] that paints [Splash]es.
+///
+/// Splash effects can be painted on an ancestor [Material] widget by obtaining
+/// an instance of this controller via [Material.of] and calling [addSplash].
+class SplashController extends RenderProxyBox {
+  /// Creates a splash controller.
+  SplashController({required this.vsync, this.color}) : super(null);
 
-  // This class should exist in a 1:1 relationship with a MaterialState object,
-  // since there's no current support for dynamically changing the ticker
-  // provider.
-  @override
+  /// The [TickerProvider] used by this controller.
+  ///
+  /// [Splash]es added to this controller with [addSplash] use this vsync
+  /// to drive their animations.
   final TickerProvider vsync;
 
-  // This is here to satisfy the MaterialInkController contract.
-  // The actual painting of this color is done by a Container in the
-  // MaterialState build method.
-  @override
+  /// (Optionally) specifies the color of the surface on which
+  /// splashes are being painted.
```

Comment:
I don't remember seeing a lot of parentheses in Flutter docs so I suggest removing them, which reads fine too.
```suggestion
  /// Optionally specifies the color of the surface on which
  /// splashes are being painted.
```
The same for the one below.